docs/DATA_FLOW.md

```diff
@@ -0,0 +1,629 @@
+  string componentClass = 3;                   // Component type (e.g., "GPU", "SYSLOG")
+  string nodeName = 13;                        // Kubernetes node name
+  
+  // Health status
+  string checkName = 4;                        // Specific check (e.g., "XID_ERROR")
+  bool isFatal = 5;                           // Critical failure
+  bool isHealthy = 6;                         // Current health state
+  string message = 7;                         // Human-readable description
+  
+  // Classification
+  RecommendedAction recommendedAction = 8;    // What should be done
+  repeated string errorCode = 9;              // Error identifiers (e.g., ["XID-48"])
+  repeated Entity entitiesImpacted = 10;      // Affected resources
+  
+  // Metadata
+  map<string, string> metadata = 11;          // Key-value pairs (GPU UUID, driver version, etc.)
+  google.protobuf.Timestamp generatedTimestamp = 12;
+  
+  // Behavior overrides
+  BehaviourOverrides quarantineOverrides = 14;
+  BehaviourOverrides drainOverrides = 15;
+}
+
+enum RecommendedAction {
+  NONE = 0;
+  COMPONENT_RESET = 2;
+  CONTACT_SUPPORT = 5;
+  RESTART_VM = 15;
+  RESTART_BM = 24;
+  REPLACE_VM = 25;
+  UNKNOWN = 99;
+}
+
+message Entity {
+  string entityType = 1;    // e.g., "GPU", "NODE", "POD"
+  string entityValue = 2;   // e.g., GPU UUID, node name
+}
+```
+
+### Example HealthEvent: GPU XID Error
+
+```json
+{
+  "version": 1,
+  "agent": "gpu-health-monitor",
+  "componentClass": "GPU",
+  "checkName": "XID_ERROR_48",
+  "isFatal": true,
+  "isHealthy": false,
+  "message": "GPU 0 reported XID 48 (Double Bit ECC Error)",
+  "recommendedAction": "REPLACE_VM",
+  "errorCode": ["XID-48"],
+  "entitiesImpacted": [
+    {
+      "entityType": "GPU",
+      "entityValue": "GPU-12345678-abcd-1234-abcd-123456789abc"
+    }
+  ],
+  "metadata": {
+    "gpu_uuid": "GPU-12345678-abcd-1234-abcd-123456789abc",
+    "gpu_index": "0",
+    "driver_version": "535.104.05",
+    "severity": "CRITICAL"
+  },
+  "generatedTimestamp": "2025-10-28T10:15:30Z",
+  "nodeName": "gpu-node-42",
+  "quarantineOverrides": null,
+  "drainOverrides": null
+}
+```
+
+---
+
+## Component Data Flow
+
+### 1. GPU Health Monitor
+
+**What it captures:**
+- GPU temperature, power, utilization (via DCGM)
+- ECC errors (single-bit, double-bit)
+- XID errors (hardware faults)
+- GPU throttling events
+
+**What it emits:**
+- `HealthEvent` via gRPC to Platform Connectors
+- Metrics to Prometheus (separate path)
+
+**Example flow:**
+```
+DCGM reports XID 48 on GPU 0
+  ↓
+Monitor creates HealthEvent:
+  - agent: "gpu-health-monitor"
+  - componentClass: "GPU"
+  - checkName: "XID_ERROR_48"
+  - isFatal: true
+  - recommendedAction: REPLACE_VM
+  - errorCode: ["XID-48"]
+  ↓
+Sends via gRPC: HealthEventOccurredV1(HealthEvents)
+```
+
+### 2. Syslog Health Monitor
+
+**What it captures:**
+- Kernel panics from journalctl
+- Hardware errors (MCE, PCIe)
+- Driver crashes
+- Out-of-memory (OOM) events
```

Comment:
It doesn't do any of this 😅 . 

It only does XID, SXID and GPU fell off the bus for now.